src/templates/plugin.main.js

```diff
@@ -200,6 +200,17 @@ export default async (context) => {
     return [null, null]
   }
 
+  const getBrowserLocale = () => {
+    if (process.client && typeof navigator !== 'undefined' && navigator.languages) {
+      // Get browser language either from navigator if running on client side, or from the headers
+      return matchBrowserLocale(locales, navigator.languages)
+    } else if (req && typeof req.headers['accept-language'] !== 'undefined') {
+      return matchBrowserLocale(locales, parseAcceptLanguage(req.headers['accept-language']))
+    } else {
+      return undefined
+    }
```

Comment:
Looks good but can we change the code so that only `string | undefined` types are possible (change `matchBrowserLocale` to return `undefined` instead of `null`)?

Reply:
I've done it this way, `matchBrowserLocale` now returns `undefined` and I've updated the tests. :)